src/proposer/proposer_logic.cpp

```diff
@@ -49,27 +49,36 @@ class LogicImpl final : public Logic {
     if (!current_tip) {
       return boost::none;
     }
+
     const blockchain::Height current_height =
         m_active_chain->GetHeight();
     const blockchain::Height target_height =
         current_height + 1;
+
+    int64_t best_time = std::max(current_tip->GetMedianTimePast() + 1, m_network->GetTime());
```

Comment:
I can not see this being updated later on (maybe I am missing it?), but it could be const then.

Reply:
I think it can, yeah.